### DISCUSSION TITLE
如何在业务类里实现前端页面的实时数据显示

### DISCUSSION THREAD

**Discussion by EchoHR910** [BYJRK/DotNet-Discussions#349] [WPF 开发]
如何在业务类里实现前端页面的实时数据显示？我目前能想到的是：写一个接口定义需要更新的数据方法，在ViewModel里实现这些更新方法，然后将ViewModel传进业务类里。我感觉这个方法太糙了，不够优雅。不知道各位大佬有无更好的实现想法

**Comment by designerlast** [comment]
建议使用weakreferenceMessager注册通知事件抽出一个服务类，UI和业务类都依赖该服务。解耦Ui和业务类
